src/trade.c

```diff
@@ -0,0 +1,4501 @@
+    /*0x0069*/ u8 unk_69;
+    /*0x006A*/ u8 filler_6A[0x6F - 0x6A];
+    /*0x006F*/ u8 unk_6F;
+    /*0x0070*/ u8 unk_70;
+    /*0x0071*/ u8 filler_71;
+    /*0x0072*/ u16 unk_72;
+    /*0x0074*/ u8 unk_74[2];
+    /*0x0076*/ u8 unk_76[2];
+    /*0x0078*/ u8 unk_78;
+    /*0x0079*/ u8 unk_79;
+    /*0x007A*/ u8 unk_7A;
+    /*0x007B*/ u8 unk_7B;
+    /*0x007C*/ u8 filler_7C[0x7E - 0x7C];
+    /*0x007E*/ u8 unk_7E;
+    /*0x007F*/ u8 filler_7F;
+    /*0x0080*/ u16 linkData[20];
+    /*0x00A8*/ u8 unk_A8;
+    /*0x00A9*/ u8 unk_A9[11];
+    /*0x00B4*/ u8 filler_B4[0x8D0-0xB4];
+    /*0x08D0*/ struct {
+        bool8 active;
+        u16 delay;
+        u8 kind;
+    } cron[4];
+    /*0x08F0*/ u16 tilemapBuffer[BG_SCREEN_SIZE / 2];
+};
+
+enum TradeStatusMsg
+{
+    TRADESTATMSG_COMMSTANDBY = 0,
+    TRADESTATMSG_CANCELED,
+    TRADESTATMSG_ONLYMON,
+    TRADESTATMSG_ONLYMON2,
+    TRADESTATMSG_WAITINGFORFRIEND,
+    TRADESTATMSG_FRIENDWANTSTOTRADE,
+    TRADESTATMSG_YOURMONCANTBETRADED,
+    TRADESTATMSG_EGGCANTBETRADED,
+    TRADESTATMSG_PARTNERMONCANTBETRADED
+};
+
+static EWRAM_DATA u8 *sSpriteTextTileBuffer = NULL;
+static EWRAM_DATA u8 *sSpriteTextTilePtrs[14] = {};
+EWRAM_DATA struct MailStruct gLinkPartnerMail[6] = {};
+EWRAM_DATA u8 gSelectedTradeMonPositions[2] = {0};
+static EWRAM_DATA struct TradeMenuResources * sTradeMenuResourcesPtr = NULL;
+
+static void sub_804C728(void);
+static void sub_804D4F8(void);
+static void sub_804D638(void);
+static void sub_804D694(u8 state);
+static void sub_804D764(void);
+static u8 shedinja_maker_maybe(void);
+static void sub_804DFF0(void);
+static void RunTradeMenuCallback(void);
+static void sub_804EAAC(u8 a0);
+static void sub_804EAE4(u8 side);
+static u8 sub_804EE6C(u8 *str, u8 whichParty, u8 partyIdx);
+static void sub_804EED4(u8 *str, u8 whichParty, u8 partyIdx);
+static void sub_804F020(u8 side);
+static void sub_804F08C(u8 a0, u8 partyIdx, u8 a2, u8 a3, u8 a4, u8 a5);
+static void sub_804F284(u8 side);
+static void sub_804F3B4(void);
+static void sub_804F3C8(u8 a0);
+static void TradeMenuAction_Summary(u8 taskId);
+static void TradeMenuAction_Trade(u8 taskId);
+static void ScheduleLinkTaskWithDelay(u16 delay, u8 kind);
+static void RunScheduledLinkTasks(void);
+static void PrintTradeErrorOrStatusMessage(u8 str_idx);
+static bool8 sub_804F610(void);
+static void RenderTextToVramViaBuffer(const u8 *name, u8 *a1, u8 unused);
+static void sub_804F748(u8 side);
+static void sub_804F890(u8 side);
+static void sub_804F964(void);
+static void sub_804F9D8(void);
+static u32 sub_804FA14(struct Pokemon * party, int partyCount, int cursorPos);
+
+static const size_t gUnknown_8260814[] = {
+    sizeof(struct SaveBlock2),
+    sizeof(struct SaveBlock1),
+    sizeof(struct MapData),
+    0x530, // unk
+    0x34, // unk
+    sizeof(struct MailStruct),
+    sizeof(struct Pokemon),
+    0x528 // unk
+};
+
+static const u16 gTradeMovesBoxTilemap[] = INCBIN_U16("graphics/trade/moves_box_map.bin");
```

Comment:
static symbols should start with prefix `s` if documented (please also check other instances in this file)